src/vivarium_nih_us_cvd/components/effects.py

```diff
@@ -194,23 +207,28 @@ def adjust_target(index: pd.Index, target: pd.Series) -> pd.Series:
                 #   to prevent divide-by-0 errors; it does make sense also since in the
                 #   equation for the scaling factor we raise the mediator RR to a power
                 #   and 1**x is always 1 anyway.
-                # 
+                #
                 #   We also only adjust the target RR if the risk RR is not 1 (TMREL).
                 #   This is not necessarily required since that would result in delta = 0
                 #   and a scaling factor would resolve to 1 always, but it will
                 #   save some computation time.
                 not_tmrel_idx = index[
                     (unadjusted_mediator_rr != 1.0) & (unadjusted_rr != 1.0)
                 ]
-                mf = mediation_factors.loc[
-                    mediation_factors["mediator_name"] == mediator, "value"
-                ].values[0]
-                delta_mediator = np.log(
-                    mf * (unadjusted_rr.loc[not_tmrel_idx] - 1) + 1
-                ) / np.log(unadjusted_mediator_rr.loc[not_tmrel_idx])
-                scaling_factor.loc[not_tmrel_idx] *= (
-                    unadjusted_mediator_rr.loc[not_tmrel_idx] ** delta_mediator
-                )
+                if self.is_target_hf:
```

Comment:
Ah, good idea about the refactor.

@rmudambi  Regarding the filtering to RR>1, though... what about for RRbmi? In that case it mathematically works (unlike the other equations), but it becomes a fundamentally different thing then since for stroke and MI we do NOT adjust if the risk RR (not just the mediator RR) is 1.

Reply:
Sorry, you're right. We need to filter out the case where the unadjusted BMI == 1, but the reason is different in the HF case than what you've described in the comment above.